src/mds/Server.cc

```diff
@@ -1720,7 +1720,9 @@ void Server::handle_osd_map()
    * using osdmap_full_flag(), because we want to know "is the flag set"
    * rather than "does the flag apply to us?" */
   mds->objecter->with_osdmap([this](const OSDMap& o) {
-      is_full = o.test_flag(CEPH_OSDMAP_FULL);
+      std::set<int64_t> full, backfillfull, nearfull;
+      o.get_full_pools(mds->cct, &full, &backfillfull, &nearfull);
```

Comment:
this is used by the mon to set/clear the pool flags (and is slow).  we only need to watch the flags.